http-client/test-nonet/Network/HTTP/ClientSpec.hs

```diff
@@ -28,13 +36,13 @@ redirectServer inner = bracket
         (N.runTCPServer (N.serverSettingsTCPSocket lsocket) app)
         (const $ inner port)
   where
-    app ad = do
-        _ <- forkIO $ forever $ N.appRead ad
-        forever $ do
+    app ad = Async.race_
+        (forever (N.appRead ad))
+        (silentIOError $ forever $ do
```

Comment:
It is described briefly in the 9a39b4b87943ec5e7fdffe3292195dd8bcf51caa commit message.

Basically, `forkIO $ forever (N.appRead ad)` thread is never killed, so it performs reading from already closed socket. To trigger it, add `performGC` at the end of test suit to force cached connections to be released -- as a result write loop fails (resource vanished), `app ad` exits, server socket is closed, `appRead ad` fails with either resource vanished or bad descriptor.

In any case, I'm open for better solution.